app/src/main/java/ru/spbhse/pocketmagic/NetworkController.java

```diff
@@ -0,0 +1,81 @@
+package ru.spbhse.pocketmagic;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+
+import java.nio.ByteBuffer;
+
+public class NetworkController {
+
+    private static AppCompatActivity currentActivity;
+    private static Controller currentController;
+    private static Network network;
+
+    public static Network createNetwork(GoogleSignInAccount account) {
+        network = new Network(account);
+        return network;
+    }
+
+    public static void startGame() {
+        Intent intent = new Intent(currentActivity, LoadActivity.class);
+        intent.putExtra("GameType", GameType.MULTIPLAYER);
+        currentActivity.startActivity(intent);
+        currentActivity.finish();
+    }
+
+    public static void finishGame() {
+        if (currentActivity.getClass() == GameActivity.class) {
```

Comment:
I would do 

if (currentActivity instanceof GameActivity)

but this is optional. Does it ever make sense to include classes that inherit from GameActivity?

Reply:
Yes, it is. The reason is when we use Google Play API in network code we have to know current activity.